com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeGIBaking.cs

```diff
@@ -166,40 +166,52 @@ private static void OnAdditionalProbesBakeCompleted()
 
                 for (int i = 0; i < numProbes; ++i)
                 {
-                    Vector4[] channels = new Vector4[3];
-
                     int j = bakingCells[c].probeIndices[i];
-
-                    // compare to SphericalHarmonicsL2::GetShaderConstantsFromNormalizedSH
-                    channels[0] = new Vector4(sh[j][0, 3], sh[j][0, 1], sh[j][0, 2], sh[j][0, 0]);
-                    channels[1] = new Vector4(sh[j][1, 3], sh[j][1, 1], sh[j][1, 2], sh[j][1, 0]);
-                    channels[2] = new Vector4(sh[j][2, 3], sh[j][2, 1], sh[j][2, 2], sh[j][2, 0]);
+                    SphericalHarmonicsL2 shv = sh[j];
 
                     // It can be shown that |L1_i| <= |2*L0|
                     // Precomputed Global Illumination in Frostbite by Yuriy O'Donnell.
                     // https://media.contentapi.ea.com/content/dam/eacom/frostbite/files/gdc2018-precomputedgiobalilluminationinfrostbite.pdf
-                    //
-                    // So divide by L0 brings us to [-2, 2],
-                    // divide by 4 brings us to [-0.5, 0.5],
-                    // and plus by 0.5 brings us to [0, 1].
-                    for (int channel = 0; channel < 3; ++channel)
+                    for (int rgb = 0; rgb < 3; ++rgb)
                     {
-                        var l0 = channels[channel][3];
+                        var l0 = sh[j][rgb, 0];
 
-                        if (l0 != 0.0f)
-                        {
-                            for (int axis = 0; axis < 3; ++axis)
-                            {
-                                channels[channel][axis] = channels[channel][axis] / (l0 * 4.0f) + 0.5f;
-                                Debug.Assert(channels[channel][axis] >= 0.0f && channels[channel][axis] <= 1.0f);
-                            }
-                        }
+                        if (l0 == 0.0f)
+                            continue;
+
+                        // TODO: Enlarging range by 2 because we're using irradiance probes. Need to derive correct factors and remove this.
+                        // Compare ProbeVolume.hlsl
+                        l0 *= 2.0f;
+
+                        // L_1^m
+                        shv[rgb, 1] = sh[j][rgb, 1] / (l0 * 2.0f) + 0.5f;
```

Comment:
Again, I'd add comments to specify what is going on (the link above only shows L1)

Reply:
we should define fully the constant: 0.9375f

for example in AmbientProbeConvolutoin.compute we do
#define fC0 (rsqrt(PI * 4.0) * rsqrt(PI * 4.0))  // Equivalent (0.282095 * (1.0 / (2.0 * sqrtPI)))
#define fC1 (rsqrt(PI * 4.0 / 3.0) * rsqrt(PI * 3.0)) // Equivalent to (0.488603 * (sqrt ( 3.0) / ( 3.0 * sqrtPI)))
#define fC2 (rsqrt(PI * 4.0 / 15.0) * rsqrt(PI * 64.0 / 15.0)) // Equivalent to (1.092548 * (sqrt (15.0) / ( 8.0 * sqrtPI)))
#define fC3 (rsqrt(PI * 16.0 / 5.0) * rsqrt(PI * 256.0 / 5.0)) // Equivalent to (0.315392 * (sqrt ( 5.0) / (16.0 * sqrtPI)))
#define fC4 (rsqrt(PI * 16.0 / 15.0) * rsqrt(PI * 256.0 / 15.0)) // Equivalent to  (0.546274 * 0.5 * (sqrt (15.0) / ( 8.0 * sqrtPI)))

help a lot to understand where the constant come from.

Reply:
Done and done